citystatmainwindow.cpp

```diff
@@ -1,69 +1,66 @@
+    m_statusBarWidget.changePopulationStatisticsMessage(static_cast<IAggragatedDataProvider*>(&m_cityStatisticsModel));
 
-    //table view style
-    ui->cityTableView->setStyleSheet("QHeaderView::section { background-color:#e1e1e1; border: 1px solid #6c6c6c; padding-left: 6px;padding-right: 6px;}");
+    ui->cityTableView->setModel(&m_cityStatisticsModel);
+    ui->cityTableView->setItemDelegateForColumn(0,&m_comboDelegate);
     ui->cityTableView->resizeColumnsToContents();
+    ui->cityTableView->setStyleSheet("QHeaderView::section { background-color:#e1e1e1;}");
 
     //connections
-    connect(m_addNewCityDialog, SIGNAL(addNewItem(QStringList)), this, SLOT(onNewItemAdded(QStringList)));
-    connect(this, SIGNAL(initiateDataSave()), m_cityStatisticsModel, SLOT(saveDataInitiated()));
-    connect(m_cityStatisticsModel, SIGNAL(populationChanged(unsigned int,unsigned int,unsigned int)),this, SLOT(updateStatusBar(unsigned int,unsigned int,unsigned int)));
-    connect(m_cityStatisticsModel, SIGNAL(isSaveDataSuccessfull(bool, unsigned int)), this, SLOT(showSaveDataResultMessage(bool, unsigned int)));
-
-    m_cityStatisticsModel->fillPopulationInfo();
+    connect(&m_cityStatisticsModel, &CityStatModel::populationStatisticsChanged, this, &CityStatMainWindow::onPopulationStatisticsChanged);
+    connect(&m_cityStatisticsModel, &CityStatModel::dataSaved, this, &CityStatMainWindow::showSaveDataResultMessage);
+    connect(ui->addNewItem, &QPushButton::clicked, this, &CityStatMainWindow::onAddNewItemClicked);
+    connect(ui->saveDataPushButton, &QPushButton::clicked, this, &CityStatMainWindow::onSaveDataPushButtonClicked);
 }
 
 CityStatMainWindow::~CityStatMainWindow()
 {
-    delete m_cityStatisticsModel;
-    delete m_comboDelegate;
-    delete m_addNewCityDialog;
     delete ui;
 }
 
-void CityStatMainWindow::onNewItemAdded(QStringList itemInfo)
-{
-    m_cityStatisticsModel->insertRows(m_cityStatisticsModel->rowCount(),1);
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,0),QVariant(itemInfo.at(0)));
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,1),QVariant(itemInfo.at(1)));
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,2),QVariant(itemInfo.at(2)));
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,3),QVariant(itemInfo.at(3)));
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,4),QVariant(itemInfo.at(4)));
-    m_cityStatisticsModel->setData(m_cityStatisticsModel->index(m_cityStatisticsModel->rowCount()-1,5),QVariant(itemInfo.at(5)));
-}
-
-void CityStatMainWindow::on_addNewItem_clicked()
+void CityStatMainWindow::onNewItemAdded(CityItem item)
 {
-    m_addNewCityDialog->show();
+    auto row = m_cityStatisticsModel.rowCount();
+    m_cityStatisticsModel.insertRows(row,1);
```

Comment:
This is not how it should work. You should insert new item into CityStatModel::m_cityItemsVector and update it, so you most likely do not even this CityStatMainWindow::onNewItemAdded slot. View is representing data from the model, when new data comes in you should only insert it into the model.